skimage/feature/peak.py

```diff
@@ -23,7 +23,10 @@ def _get_high_intensity_peaks(image, mask, num_peaks, min_distance, p_norm):
     else:
         max_out = None
 
-    coord = ensure_spacing(coord, spacing=min_distance, p_norm=p_norm, max_out=max_out)
+    if min_distance > 1:
+        coord = ensure_spacing(
+            coord, spacing=min_distance, p_norm=p_norm, max_out=max_out
+        )
```

Comment:
Is it safe to skip for all p-norms? If so, why not add this performance friendly code path directly to `ensure_spacing`?

Reply:
I initially did try adding it to ensure_spacing, but it is more general in that the coordinates don't have to correspond to pixel locations. It also has test cases that involve duplicate points. Given that, I think it is only safe to do for peak_local_max where points correspond to unique pixel coordinates.

It should be correct for all p-norms as the closest spacing is 1 pixel in the horizontal or vertical direction regardless of which norm is used.